jvm-libs/linea/clients/linea-l1-contract-client/src/main/kotlin/build/linea/contract/l1/Web3JLineaRollupSmartContractClientReadOnly.kt

```diff
@@ -63,8 +63,7 @@ open class Web3JLineaRollupSmartContractClientReadOnly(
     } as T
   }
 
-  protected val smartContractVersionCache: AtomicReference<LineaContractVersion> =
-    AtomicReference(fetchSmartContractVersion().get())
+  private val smartContractVersionCache = AtomicReference<LineaContractVersion>(null)
```

Comment:
Changed this to private and being initially assigned to null to avoid throwing connectivity errors upon class initialization such as when the target rpc node was not ready to be connected. 
This actually happens when I switched to use `Web3JLineaRollupSmartContractClientReadOnly` in `LineaBesuEngineBlockTagUpdater` (part of this PR) during a local stack run, given that  `Web3JLineaRollupSmartContractClientReadOnly` was trying to make a call to get the contract version from` l1-el-node` at a very early stage during class initialization when the node was not ready to be connected.

Reply:
I am ok with that. Did you test if the smart contract upgrade changes still work?